### PR TITLE
Development docs

### PR DESCRIPTION
**Pull request by regDaniel**
**Summary**

An additional section "Roadmap to your first contribution" should provide some more guidance on where and how to start implementing. + some additional text to avoid pitfalls with `mamba` and `pre-commit` and a git disclaimer for users not familiar with the git workflow at APN.

### CODE REVIEW

**Comment by regDaniel** [comment]
launch jenkins

File: tmpl/README.md.j2 [hunk 1]
```diff
@@ -12,25 +12,54 @@ and have no additional dependencies. Moreover we distinguish pinned installation
...
 every time you add new imports while developing. Check the next section to find some guidance on the development process if you are new to Python and/ or APN.
 
 
+### Roadmap to your first contribution
+
+Generally, the source code of your library is located in `src/<library_name>`. The blueprint will generate some example code in `mutable_number.py`, `utils.py` and `cli.py`. `cli.py` thereby serves as an entry
+point for functionalities you want to execute from the command line, it is based on the Click library. If you do not need interactions with the command line, you should remove `cli.py`. Moreover, of course there exist other options for command line interfaces,
+a good overview may be found here (https://realpython.com/comparing-python-command-line-parsing-libraries-argparse-docopt-click/), we recommend however to use click. The provided example
+code should provide some guidance on how the individual source code files interact within the library. In addition to the example code in `src/<library_name>`, there are examples for
```
Comment (Itschotsch) [inline tmpl/README.md.j2:42]: Wording:
> The **provided** example code should **provide**

File: tmpl/README.md.j2 [hunk 2]
```diff
@@ -12,25 +12,54 @@ and have no additional dependencies. Moreover we distinguish pinned installation
 is based on first level dependencies listed in `requirements/requirements.yml` and `requirements/dev-requirements.yml` respectively. If you start developing, you might want to do a free
 development installation and export the environments right away.
 ```bash
-setup_env.sh -d -e -m -n <package_env_name>
+setup_env.sh -u -d -e -n <package_dev_env_name>
```
Comment (Itschotsch) [inline tmpl/README.md.j2:15]: Maybe mention the following situation just to save the user some time. I ran into this issue several times with completely independent installations:
```
An error has occurred: InvalidManifestError: 
==> File /users/jappel/.cache/pre-commit/repozhhauwxe/.pre-commit-hooks.yaml
==> At Hook(id='black-jupyter')
==> At key: types_or
==> At index 2
=====> Type tag 'jupyter' is not recognized.  Try upgrading identify and pre-commit?
Check the log at /users/jappel/.cache/pre-commit/pre-commit.log
```
This occured after running `setup_env.sh` when I wanted to run `pre-commit`. Force-re-installing `pre-commit` fixed it for me every time:
```bash
pip install --force-reinstall -U pre-commit
```
If this is not a bug that can be fixed easily, I would add a small note here, mentioning that if one runs into this issue, this is the way to fix it.

**Review by Itschotsch — APPROVED** [review]

**Comment by regDaniel** [comment]
launch jenkins

File: docs/faq.rst [hunk 3]
```diff
@@ -43,6 +43,21 @@ navigating to your package root directory and typing :code:`pre-commit run --all
 on how to run the tools from the command line.
 
 
+**Q:** Pre-commit fails with the following error:
+
+..code-block:: bash
+
+    An error has occurred: InvalidManifestError:
+    ==> File /users/jappel/.cache/pre-commit/repozhhauwxe/.pre-commit-hooks.yaml
```
Comment (Itschotsch) [inline docs/faq.rst:51]: Maybe make it more general:
`/users/<username>/.cache/pre-commit/repozhhauwxe/.pre-commit-hooks.yaml`

**Comment by regDaniel** [comment]
sorry, I've missed this one. Will do.

**Pull request merged**